### PR TITLE
NOISSUE: generalize slotdebugger predicates

### PR DESCRIPTION
**Pull request by MikhailZhukov**
**- What I did**
Allow more flexible usage debugger predicates 

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/testutils/journal/journal.go [hunk 1]
```diff
@@ -63,24 +63,24 @@ func (p *Journal) Subscribe(outFn predicate.SubscriberFunc) {
 	p.dispenser.Subscribe(outFn)
 }
 
-func (p *Journal) WaitOnce(pFn predicate.Func) synckit.SignalChannel {
+func (p *Journal) WaitOnce(pFn predicate.Predicate) synckit.SignalChannel {
```
Comment (cyraxred) [inline ledger-core/testutils/journal/journal.go:66]: Bad naming - predicate.Func is better

File: ledger-core/testutils/predicate/predicate.go [hunk 2]
```diff
@@ -9,25 +9,27 @@ import (
 	"github.com/insolar/assured-ledger/ledger-core/testutils/debuglogger"
 )
 
-func Never() Func {
+type Predicate = func(debuglogger.UpdateEvent) bool
```
Comment (cyraxred) [inline ledger-core/testutils/predicate/predicate.go:12]: Do NOT name type same as package (except for very rare exceptions)

File: ledger-core/testutils/predicate/predicate_slotlink.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+func NewSlotLinkFilter(slotLink smachine.SlotLink) SlotLinkFilter {
+	return SlotLinkFilter{slotLink: slotLink}
+}
+
+type SlotLinkFilter struct {
+	slotLink smachine.SlotLink
+}
+
+func (s SlotLinkFilter) IsSlotLink(event debuglogger.UpdateEvent) bool {
```
Comment (cyraxred) [inline ledger-core/testutils/predicate/predicate_slotlink.go:21]: Misguiding name, reads like "UpdateEvent is SlotLink". May be "IsFromSlot"?

Comment (MikhailZhukov) [reply ledger-core/testutils/predicate/predicate_slotlink.go]: done

File: ledger-core/testutils/predicate/predicate.go (hunk 2 above)
Comment (MikhailZhukov) [reply ledger-core/testutils/predicate/predicate.go]: done

Comment (MikhailZhukov) [reply ledger-core/testutils/predicate/predicate.go]: done

File: ledger-core/testutils/journal/journal.go (hunk 1 above)
Comment (MikhailZhukov) [reply ledger-core/testutils/journal/journal.go]: done

**Pull request merged**